src/api/crypto.ts

```diff
@@ -1,39 +1,115 @@
 import { TransactionRequest } from '@ethersproject/abstract-provider';
+import { entropyToMnemonic, HDNode } from '@ethersproject/hdnode';
+import { randomBytes } from '@ethersproject/random';
 import { Wallet } from '@ethersproject/wallet';
 import { ipcMain } from 'electron';
 
 import { ipcBridgeMain } from '@bridge';
-import { CryptoRequest, CryptoRequestType, CryptoResponse, TAddress } from '@types';
-import { addHexPrefix, generateDeterministicAddressUUID } from '@utils';
+import { MNEMONIC_ENTROPY_BYTES } from '@config';
+import {
+  CryptoRequest,
+  CryptoRequestType,
+  CryptoResponse,
+  GetMnemonicAddressArgs,
+  GetMnemonicAddressesArgs,
+  TAddress,
+  WalletType
+} from '@types';
+import { addHexPrefix, generateDeterministicAddressUUID, toChecksumAddress } from '@utils';
 
-// @todo Keystore
-
-export const sign = (wallet: Wallet, tx: TransactionRequest) => {
+const sign = (wallet: Wallet, tx: TransactionRequest) => {
   return wallet.signTransaction(tx);
 };
 
-export const signWithMnemonic = (mnemonicPhrase: string, dPath: string, tx: TransactionRequest) => {
-  return sign(Wallet.fromMnemonic(mnemonicPhrase, dPath), tx);
-};
-
-export const signWithPrivateKey = (privateKey: string, tx: TransactionRequest) => {
+const signWithPrivateKey = (privateKey: string, tx: TransactionRequest) => {
   return sign(new Wallet(addHexPrefix(privateKey)), tx);
 };
 
-export const getAddress = (privateKey: string) => {
+const getPrivateKeyAddress = (privateKey: string) => {
   const address = new Wallet(addHexPrefix(privateKey)).address as TAddress;
   return { address, uuid: generateDeterministicAddressUUID(address) };
 };
 
+const createMnemonicWallet = () => {
+  const entropy = randomBytes(MNEMONIC_ENTROPY_BYTES);
+  return entropyToMnemonic(entropy);
+};
+
+const getMnemonicAddress = ({
+  dPath,
+  phrase,
+  password
+}: {
+  dPath: string;
+  phrase: string;
+  password?: string;
+}) => {
+  const rootNode = HDNode.fromMnemonic(phrase, password);
+  const node = rootNode.derivePath(dPath);
+  const address = toChecksumAddress(node.address) as TAddress;
+  return {
+    address,
+    uuid: generateDeterministicAddressUUID(address),
+    privateKey: node.privateKey,
+    dPath
+  };
+};
+
+const getMnemonicAddresses = ({
+  dPathBase,
+  limit,
+  offset = 0,
+  phrase,
+  password
+}: {
+  dPathBase: string;
+  phrase: string;
+  password?: string;
+  offset?: number;
+  limit: number;
+}) => {
+  const rootNode = HDNode.fromMnemonic(phrase, password);
+  const addresses = [];
+  for (let i = 0; i < limit; i++) {
+    const index = i + offset;
+    const dPath = `${dPathBase}/${index}`;
+    const node = rootNode.derivePath(dPath);
```

Comment:
It may be slightly faster to derive the base master node (`dPathBase`) once, then deriving the child keys from that.

```suggestion
  const baseNode = rootNode.derivePath(basePath);
  for (let i = 0; i < limit; i++) {
    const index = i + offset;
    const node = baseNode.derivePath(`${index}`);
```